rust/cubesqlplanner/cubesqlplanner/src/planner/sql_evaluator/sql_call.rs

```diff
@@ -217,16 +220,72 @@ impl SqlCall {
                     query_tools.base_tools().security_context_for_rust()?,
                 ))
             }
-            ContextSymbolDep::FilterParams => MemberSqlArg::ContextSymbol(
-                ContextSymbolArg::FilterParams(query_tools.base_tools().filters_proxy()?),
-            ),
-            ContextSymbolDep::FilterGroup => MemberSqlArg::ContextSymbol(
-                ContextSymbolArg::FilterGroup(query_tools.base_tools().filter_group_function()?),
-            ),
+            ContextSymbolDep::FilterParams => {
+                let filters = visitor.all_filters();
+                let native_filters = self.filters_to_native_filter_item(filters);
+                let r = query_tools
+                    .base_tools()
+                    .filters_proxy_for_rust(native_filters)?;
+                MemberSqlArg::ContextSymbol(ContextSymbolArg::FilterParams(r))
+            }
+            ContextSymbolDep::FilterGroup => {
+                let filters = visitor.all_filters();
+                let native_filters = self.filters_to_native_filter_item(filters);
+                let r = query_tools
+                    .base_tools()
+                    .filter_group_function_for_rust(native_filters)?;
+                MemberSqlArg::ContextSymbol(ContextSymbolArg::FilterGroup(r))
+            }
             ContextSymbolDep::SqlUtils => MemberSqlArg::ContextSymbol(ContextSymbolArg::SqlUtils(
                 query_tools.base_tools().sql_utils_for_rust()?,
             )),
         };
         Ok(res)
     }
+
+    fn filters_to_native_filter_item(
+        &self,
+        filter: Option<Filter>,
+    ) -> Option<Vec<NativeFilterItem>> {
+        if let Some(filter) = filter {
+            let mut res = Vec::new();
+            for item in filter.items.iter() {
+                res.push(self.filters_to_native_filter_item_impl(item));
+            }
+            Some(res)
+        } else {
+            None
+        }
+    }
```

Comment:
I think this one may be rewritten in more rust-idiomatic style, smth like:
```suggestion
fn filters_to_native_filter_item(
    &self,
    filter: Option<Filter>,
) -> Option<Vec<NativeFilterItem>> {
    filter.map(|filter| 
        filter.items.iter()
            .map(|item| self.filters_to_native_filter_item_impl(item))
            .collect()
    )
}
```

Reply:
This is a very temporary solution that will be removed within a month, so I don't see the point in doing much styling here yet :)